src/exts/error_handler/error_handler.py

```diff
@@ -0,0 +1,79 @@
+import textwrap
+
+from discord import ApplicationContext, Bot, Cog, Colour, Embed, errors
+from discord.ext.commands import errors as commands_errors
+
+from src.settings import GITHUB_REPO
+from src.utils.log import get_logger
+
+log = get_logger(__name__)
+
+
+class ErrorHandler(Cog):
+    """Cog to handle any errors invoked from commands."""
+
+    def __init__(self, bot: Bot) -> None:
+        self.bot = bot
+
+    async def send_error_embed(
+        self,
+        ctx: ApplicationContext,
+        *,
+        title: str | None = None,
+        description: str | None = None,
+    ) -> None:
+        """Send an embed regarding the unhandled exception that occurred."""
+        if title is None and description is None:
+            raise ValueError("You need to provide either a title or a description.")
+
+        embed = Embed(title=title, description=description, color=Colour.red())
+        await ctx.respond(f"Sorry, {ctx.author.mention}", embed=embed)
+
+    async def send_unhandled_embed(self, ctx: ApplicationContext, exc: BaseException) -> None:
+        """Send an embed regarding the unhandled exception that occurred."""
+        msg = f"Exception {exc!r} has occurred from command {ctx.command.qualified_name} invoked by {ctx.author.id}"
+        if ctx.message:
+            msg += f" in message {ctx.message.content!r}"
+        if ctx.guild:
+            msg += f" on guild {ctx.guild.id}"
+        log.warning(msg)
+
+        await self.send_error_embed(
+            ctx,
+            title="Unhandled exception",
+            description=textwrap.dedent(
+                f"""
+                Unknown error has occurred without being properly handled.
+                Please report this at the [GitHub repository]({GITHUB_REPO})
+
+                **Command**: `{ctx.command.qualified_name}`
+                **Exception details**: ```{exc.__class__.__name__}: {exc}```
+                """
+            ),
+        )
+
+    @Cog.listener()
+    async def on_application_command_error(self, ctx: ApplicationContext, exc: errors.DiscordException) -> None:
+        """Handle exceptions that have occurred while running some command."""
+        if isinstance(exc, errors.CheckFailure | commands_errors.CheckFailure):
+            if isinstance(exc, commands_errors.NotOwner):
```

Comment:
`CheckFailure` isn't supposed to mean missing permissions. Instead should use https://docs.pycord.dev/en/stable/ext/commands/api.html#discord.ext.commands.MissingPermissions, also, why a nested if for NotOwner since it inherits from Check faliure?

I would do 
```
if isinstance NotOwner
    yada yada
    return
if isinstance MissingPermissions
    yada yada
    return

Reply:
> since it inherits from Check faliure

exactly, I figured it would be cleaner to have the errors of the same kind close groupped under the same parent if.

> Instead should use https://docs.pycord.dev/en/stable/ext/commands/api.html#discord.ext.commands.MissingPermissions,

Should it though? This error is only raised when the callers is missing a particular discord permission, for example `MissingRole` is yet another `CheckError` subclass that also essentially implies the caller doesn't have a right to run this command, which is what I intended behind that error message.

Is there a case where a `CheckError` really doesn't have anything to do with the rights to run that command and we're instead checking something else?

Reply:
Well an example that comes to my mind is `BotMissingPermissions` which means the bot is incorrectly set up probably.

After checking the source, there is too `NoPrivateMessage`, `PrivateMessageableOnly` and others. Maybe, theese can be caught *first*, so that allt he others imply the user is missing permissions... idk, theese are errors for default checks, probably we will implement our own checks at some point, which will mean have our own errors caught.